src/writers/smt/writer.rs

```diff
@@ -638,9 +645,15 @@ impl<'a> CompositionSmtWriter<'a> {
         ident: &Identifier,
         opt_idx: &Option<Expression>,
         sample_id: &Option<usize>,
+        sample_name: &Option<String>,
         tipe: &Type,
     ) -> SmtExpr {
         let sample_id = sample_id.expect("found a None sample_id");
+        let sample_identifier = if let Some(sname) = sample_name {
+            sname
+        } else {
+            &format!("{}", sample_id)
+        };
```

Comment:
let's .expect() that this is set and pull the sample position out of the sample info (which should be in self or oracle_ctx), using the sample_id. debug assert that sample_name matches what is in the sample position.